modules/c++/sys/source/OSWin32.cpp

```diff
@@ -221,14 +222,21 @@ std::string sys::OSWin32::getEnv(const std::string& s) const
 
     // If we can use a normal size buffer, lets not bother to malloc
 
-    char *buffer = new char[size + 1];
-    DWORD retVal = GetEnvironmentVariable(s.c_str(), buffer, size);
-    result = buffer;
-    delete [] buffer;
-
+    std::vector<char> buffer;
+    buffer.resize(size+1);
+    DWORD retVal = GetEnvironmentVariable(s.c_str(), static_cast<char*>(&buffer[0]), size);
```

Comment:
Don't need the `static_cast` here or in the next line... `&buffer[0]` is a `char*`